google/cloud/storage/tests/object_file_integration_test.cc

```diff
@@ -625,6 +625,51 @@ TEST_F(ObjectFileIntegrationTest, UploadPortionRegularFile) {
   EXPECT_EQ(0, std::remove(file_name.c_str()));
 }
 
+TEST_F(ObjectFileIntegrationTest, ResumableUploadFileCustomHeader) {
+  // Test relies on emulator for capturing custom header
+  if (!UsingEmulator()) GTEST_SKIP();
+ 
+  StatusOr<Client> client = MakeIntegrationTestClient();
+  ASSERT_STATUS_OK(client);
+ 
+  auto file_name = ::testing::TempDir() + MakeRandomFilename();
+  auto object_name = MakeRandomObjectName();
+ 
+  // We will construct the expected response while streaming the data up.
+  std::ostringstream expected;
+  // Create a file with the contents to upload.
+  std::ofstream os(file_name, std::ios::binary);
+  WriteRandomLines(os, expected);
+  os.close();
+ 
+  // Create a CustomHeader object to send in request.
+  CustomHeader custom_header("custom_header_key","custom_header_value");
+ 
+  StatusOr<ObjectMetadata> meta = client->UploadFile(
+      file_name, bucket_name_, object_name, custom_header,
+      IfGenerationMatch(0), NewResumableUploadSession());
+ 
+  ASSERT_STATUS_OK(meta);
+  EXPECT_EQ(object_name, meta->name());
+  EXPECT_EQ(bucket_name_, meta->bucket());
+  auto expected_str = expected.str();
+  ASSERT_EQ(expected_str.size(), meta->size());
+ 
+  ASSERT_TRUE(meta->has_metadata("x-goog-emulator-custom-header"));
+  EXPECT_EQ("custom_header_value", meta->metadata("x-goog-emulator-custom-header"));
```

Comment:
Ahh sorry, I misunderstood the code :sweat_smile:. One more comment, we have to wrap this code around
```
if (UsingEmulator())
```

Reply:
Sorry to butt in, but it seems strange to have marked this comment as off-topic. "Mark as obsolete" would've made more sense here.

Reply:
Hmm... yeah, you are right.  OTOH, I do not recall explicitly marking it as "off-topic" and/or "obsolete" or anything else.  Is this a new feature that I am misusing or something?

Reply:
No I think it's not a new feature but it's because we rarely use it. I think we usually mark a review as resolve. Hiding a review is when I feel that there is a serious problem with it. In addition, Github does not have "obsolete" option indeed.

<img width="565" alt="image" src="https://user-images.githubusercontent.com/17511625/102252401-a1c9af80-3f38-11eb-9191-0030b7b6020e.png">

Reply:
Yeah "obsolete" was a freudian slip. "Resolved" is what I mean. But "off-topic" makes it sound like the comment is disruptive (when it's clearly not).

Reply:
> But "off-topic" makes it sound like the comment is disruptive (when it's clearly not).

Ah I did not think about it. But there is a "Spam" option for that kind of comments though. And "Resolved" is not really this case so I just go with "off-topic".